### ISSUE TITLE
RFE: user configs from «--user-package-configuration-file» should overwrite hardcoded standard.nuitka-package.config.yml

### ISSUE THREAD

**Issue by belonesox** [Nuitka/Nuitka#2325]
User configs from «--user-package-configuration-file» should overwrite hardcoded `standard.nuitka-package.config.yml`, 
may be with specific 'overload' flag.

There are situations then user have to adapt how to find dependent DLLs/SO even for packages known to Nuitka.
Better to do it without patching/forking Nuitka code.
Unfortunately, merging of config files disables redefining config keys («TODO» waiting...).

See `nuitka/utils/Yaml.py`

```python
    def update(self, other):
        # TODO: Full blown merging, including respecting an overload flag, where a config
        # replaces another one entirely, for now we expect to not overlap.
        for key, value in other.items():
            assert key not in self.data, key

            self.data[key] = value
```

**Comment by kayhayen** [comment]
Yes, that merge implementation is something that I could use help with. I have not yet considered if it is even doable always, what should be added, what should be replaced, and what order to read files in.

I would love to split up the standard config by topic. So that e.g. IPython related stuff is only in a single file. 

Also, how would a user config disable something that the standard config says?

Upstream packages of Yaml do not offer ways of merging if I recall correctly, so we would be on our own here. And then, I would want to trace the origin of data too, such that in error messages, we can point to the correct file....

**Comment by belonesox** [comment]
Yes, my point is that «user configuration files» are always needed to customize or fix rare use cases without resorting to hacking the code. If they are not available, then hacking, forking, or monkey patching would be better options.

And mostly always, if we have different level of configs («System», «User», «Project», ... ), lower levels of configs should override high level (I cannot remember exceptions from this rule).

About this case, I can imagine other use cases, such as a user who has built several versions of a particular SO/DLL library locally (for testing, research, etc.) and wants to test Nuitka builds with different user package configurations to compare, test, and benchmark these builds.

**Comment by kayhayen** [comment]
I am just not seeing how 2 levels of dll configurations would merge successfully. I am aware that there will have to be a priority of course. These DLL configs can easily conflict due to replacement intentions and amend an existing one.

I believe the merging algorithm would have to be non-trivial. What if you mean to disable a replacement in anti-bloat or add one, or add a condition to an existing one, i.e. replace one.

This feature is precisely not implemented yet, because so far I have been too stupid to do it.

It might just be the case, that we have to leave user configuration for packages that have no "official" configuration provided.

As an aside, there won't be user settings for Nuitka, at least not in my plans. What I call user config is user provided config for that specific compilation.

**Comment by Maxwell175** [comment]
I think it is best to keep it simple. Since overriding the package config is always considered an advanced use case, it should simply replace all of the existing configuration for the module. It should be up to the user config to keep any standard settings that are deemed necessary.

**Comment by kayhayen** [comment]
I believe I will implement a merge for things only where it's trivial to do at first at least. Then give up on everything else. Also, the merging should result in some sort of ability to produce an updated standard file that contains the user change as well to make it a good workflow for PRs. For DLLs I might not make a merge, maybe other things not. But this is definitely in no way a priority for me. I might start it when I encounter an updating situation and feel like it at some point.
